src/Network/Probe.php

```diff
@@ -738,19 +739,19 @@ private static function detect(string $uri, string $network, int $uid, array $ap
 
 		$result = [];
 
-		if (in_array($network, ["", Protocol::DFRN])) {
+		if (in_array($network, ['', Protocol::DFRN])) {
 			$result = self::dfrn($webfinger);
 		}
-		if ((!$result && ($network == "")) || ($network == Protocol::DIASPORA)) {
+		if (!$result && in_array($network, ['', Protocol::DIASPORA])) {
```

Comment:
The new condition isn't equivalent to the previous one, is it intended?

With:
- `A`: `$result`
- `B`: `$network == ''`
- `C`: `$network == Protocol::DIASPORA`

Former condition: `(NOT(A) AND B) OR C`
New condition: `NOT(A) AND (B OR C)`

Reply:
It doesn't matter here. I'm on my mobile, so I keep the answer short.

Reply:
Thanks, I just wanted to make sure.

Reply:
`$result` is empty both when the previous check wasn't successful (which would only be executed when the network had been empty) or when we would check for a dedicated network.